ci: Drop no longer needed `macos_sdk_cache`

hebasto opened the pull request:
It has been cached in the Docker image since https://github.com/bitcoin/bitcoin/pull/27028.

maflcko reviewed (commented):
lgtm. Can use `GLOBAL_TASK_TEMPLATE` now?

hebasto committed:
ci: Drop no longer needed `macos_sdk_cache`

It has been cached in the Docker image since https://github.com/bitcoin/bitcoin/pull/27028.

hebasto commented:
> Can use `GLOBAL_TASK_TEMPLATE` now?

Thanks! Updated.
